src/esp/gfx/RenderCamera.cpp

```diff
@@ -39,10 +43,46 @@ RenderCamera& RenderCamera::setProjectionMatrix(int width,
   return *this;
 }
 
-RenderCamera& RenderCamera::draw(MagnumDrawableGroup& drawables) {
-  // TODO: visibility culling
-  MagnumCamera::draw(drawables);
-  return *this;
+uint32_t RenderCamera::draw(MagnumDrawableGroup& drawables) {
+  uint32_t numDrawnObjects = drawables.size();
+  if (!frustumCulling) {
+    MagnumCamera::draw(drawables);
+  } else {
```

Comment:
```cpp
if (!frustumCulling) {
  MagnumCamera::draw(drawables);
  return drawables.size();
}
```

then the (longer) rest of the function doesn't need to be indented and the branching flow is clearer (to me at least).